libraries/databag.rb

```diff
@@ -73,27 +76,24 @@ def self.to_value(array, _options = {}) # rubocop:disable CyclomaticComplexity,
     # If provided nil or a string that doesn't resolve to a data_bag + item at least will return nil
     def self.load(string, options = {}) # rubocop:disable CyclomaticComplexity, PerceivedComplexity
       opts = {
-        type: :simple,
         pick_context: nil,
         handle_load_failure: false
       }.merge(options)
-      clazz =
-        case opts[:type]
-        when :simple
-          Chef::DataBagItem
-        when :vault
-          require 'chef-vault'
-          ChefVault::Item
-        else
-          fail "Unexpected type of DataBag #{opts[:type]}"
-        end
 
       data_bag, bag_item, key = to_a(string, options)
       value =
         if data_bag && bag_item
           # Exception handler to check if data_bag or data_bag_item exists
           begin
-            bag = clazz.load(data_bag, bag_item)
+            bag = case ChefVault::Item.data_bag_item_type(data_bag, bag_item)
+            when :normal
+              data_bag_item(data_bag, bag_item)
+            when :encrypted
+              secret = ::IO.read(opts[:secret]) if opts[:secret]
+              data_bag_item(data_bag, bag_item, secret)
+            when :vault
+              ChefVault::Item.load(data_bag, bag_item)
```

Comment:
I tried to use the chef_vault_item helper method from the chef-vault cookbook here, and it worked fine in the recipe specs and on vagrant.  However, I couldn't get it working in the library rspecs here since it wasn't loading the libraries directory from the chef-vault cookbook.